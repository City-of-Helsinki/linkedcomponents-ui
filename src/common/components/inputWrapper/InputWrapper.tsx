import classNames from 'classnames';
import React from 'react';

import FieldLabel from '../fieldLabel/FieldLabel';
import styles from './inputWrapper.module.scss';

type InputWrapperProps = {
  children?: React.ReactNode;
  className?: string;
  hasIcon?: boolean;
  helperText?: string;
  hideLabel?: boolean;
  id: string;
  invalid?: boolean;
  label?: string | React.ReactNode;
  labelText?: string;
  required?: boolean;
  style?: React.CSSProperties;
  tooltipLabel?: string;
  tooltipText?: string;
  tooltipButtonLabel?: string;
};

const InputWrapper = ({
  children,
  className = '',
  hasIcon = false,
  helperText,
  hideLabel = false,
  id,
  invalid = false,
  label,
  labelText,
  required = false,
  style,
  tooltipLabel,
  tooltipText,
  tooltipButtonLabel,
}: InputWrapperProps) => (
  <div
    className={classNames(
      styles.root,
      { [styles.hasIcon]: hasIcon, [styles.invalid]: invalid },
      className
    )}
    style={style}
  >
    {(label || labelText) && (
      <FieldLabel
        inputId={id}
        hidden={hideLabel}
        label={label || labelText}
        required={required}
        tooltipLabel={tooltipLabel}
        tooltipButtonLabel={tooltipButtonLabel}
        tooltipText={tooltipText}
      />
    )}
    <div className={classNames(styles.inputWrapper)}>{children}</div>
    {helperText && <div className={styles.helperText}>{helperText}</div>}
  </div>
);

export default InputWrapper;
