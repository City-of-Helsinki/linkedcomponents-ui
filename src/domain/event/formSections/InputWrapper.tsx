import classNames from 'classnames';
import capitalize from 'lodash/capitalize';
import React from 'react';

import { INPUT_MAX_WIDTHS } from '../../../constants';
import styles from './inputWrapper.module.scss';

type Columns = 5 | 6 | 7 | 8 | 9 | 10;
type InputColumns = 4 | 5 | 6 | 7 | 8;
type MaxWidth = INPUT_MAX_WIDTHS.MEDIUM | INPUT_MAX_WIDTHS.LARGE;
type Offset = 0 | 1 | 2;

type Props = {
  button?: React.ReactNode;
  className?: string;
  columns: Columns;
  inputColumns: InputColumns;
  maxWidth?: MaxWidth;
  offset?: Offset;
};

const InputWrapper: React.FC<Props> = ({
  button,
  children,
  className,
  columns,
  inputColumns,
  maxWidth,
  offset = 0,
}) => {
  return (
    <div
      className={classNames(
        styles.inputWrapper,
        styles[`columns${columns}`],
        className
      )}
    >
      <div
        className={classNames(
          styles[`inputColumns${inputColumns}`],
          styles[`offset${offset}`],
          maxWidth && styles[`maxWidth${capitalize(maxWidth)}`]
        )}
        style={{ maxWidth }}
      >
        {children}
      </div>
      {button && <div>{button}</div>}
    </div>
  );
};

export default InputWrapper;
