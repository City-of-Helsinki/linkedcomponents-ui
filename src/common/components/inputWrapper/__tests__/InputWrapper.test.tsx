import { render } from '@testing-library/react';
import React from 'react';

import InputWrapper from '../InputWrapper';

const wrapperProps = {
  helperText: 'helper text',
  label: 'label text',
  id: 'test',
};

it('matched snapshot', () => {
  const { asFragment } = render(
    <InputWrapper {...wrapperProps}>
      <input id="test" />
    </InputWrapper>
  );
  expect(asFragment()).toMatchSnapshot();
});
