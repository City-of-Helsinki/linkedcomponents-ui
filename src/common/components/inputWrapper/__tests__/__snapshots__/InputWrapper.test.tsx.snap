// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <label
      class="label "
      for="test"
    >
      label text
    </label>
    <div
      class="inputWrapper"
    >
      <input
        id="test"
      />
    </div>
    <div
      class="helperText"
    >
      helper text
    </div>
  </div>
</DocumentFragment>
`;
